ckanext/ontario_theme/templates/internal/package/resource_validation.html

```diff
@@ -23,19 +30,79 @@
     </div>
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
   {% else %}
+    {% set is_csv = "False" %}
     {% set success_message = "Data uploaded successfully" %}
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
+    <a href="{{ action }}">
+      <button id="nextBtn"
+              class="ontario-button ontario-button--primary"
+              name="save"
+              type="submit">Continue</button>
+    </a>
   {% endif %}
 
-  {% set package_id = resource.get('package_id') %}
-  {% set resource_id = resource.get('id') %}
-  {% set action = h.url_for('datastore.dictionary',
-      id=package_id,
-      resource_id=resource.id) %}
- 
-  <form method="get" action="{{ action }}">
-    <button class="ontario-button ontario-button--primary"
-            name="save"
-            type="submit">Continue</button>
-  </form>
+  {% if is_csv == "True" %}
+    <script>
+      function poll_datastore() {
+        const resId = "{{ resource.id }}";
+        const xhr = new XMLHttpRequest();
+        const url = `/api/3/action/resource_show?id=${resId}`
+
+        xhr.open("POST", url, true);
+        xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+        xhr.send();
+
+        xhr.onreadystatechange = () => {                                
+          if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+            let thisResponse = JSON.parse(xhr.response);
+
+            if (thisResponse.result.datastore_active) {
+              const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
+         
+              const fixedDelay = async () => {
+                await sleep(2000)
+                document.getElementById("pendingTxt").innerHTML = "Data file uploaded. Please click Continue.";
+                document.getElementById("spinnerSVG").style.visibility = "hidden";
+                document.getElementById("step2Btn").style.visibility = "visible";
+              }
+              fixedDelay();
+
+            } else {
+            setTimeout(poll_datastore, 1000);
+            }
+          }
+        }
+      };
+      poll_datastore();
+    </script>
+
+    <div class="ontario-small-12 ontario-row">
+      <div class="ontario-small-12 ontario-columns">
+        <div id="loadingSpinner"
+             aria-hidden='false'
+             role='alert'
+             aria-live='assertive'>
+          <div class='ontario-loading-indicator'>
+            <svg id="spinnerSVG"
+                 class='ontario-loading-indicator__spinner-small'
+                 viewBox="25 25 50 50"
+                 xmlns='http://www.w3.org/2000/svg'>
```

Comment:
Can all HTML attributes values be in double quotes please?

Reply:
Will do. I got the code from https://designsystem.ontario.ca/components/detail/loading-indicator.html, which has a mix of single and double quotes. Can you let them know?